### PR TITLE
Move initial WebRTC demo

### PR DESCRIPTION
**Pull request by jbolinger**

### CODE REVIEW

File: WebRTCSample/app/res/xml/preferences.xml [hunk 1]
```diff
@@ -0,0 +1,252 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/xml/preferences.xml:1]: nit: Missing header

File: WebRTCSample/app/res/values/styles.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/values/styles.xml:1]: nit: Missing header

File: WebRTCSample/app/res/values/strings.xml [hunk 3]
```diff
@@ -0,0 +1,230 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/values/strings.xml:1]: nit: Missing header

File: WebRTCSample/app/res/values/arrays.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/values/arrays.xml:1]: nit: Missing header

File: WebRTCSample/app/res/mipmap-anydpi-v26/ic_launcher_round.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/mipmap-anydpi-v26/ic_launcher_round.xml:1]: nit: Missing header

File: WebRTCSample/app/res/mipmap-anydpi-v26/ic_launcher.xml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/mipmap-anydpi-v26/ic_launcher.xml:1]: nit: Missing header

File: WebRTCSample/app/res/menu/connect_menu.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/menu/connect_menu.xml:1]: nit: Missing header

File: WebRTCSample/app/res/layout/glass_fragment_hud.xml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/layout/glass_fragment_hud.xml:1]: nit: Missing header

File: WebRTCSample/app/res/layout/glass_fragment_call.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/layout/glass_fragment_call.xml:1]: nit: Missing header

File: WebRTCSample/app/res/layout/glass_activity_connect.xml [hunk 10]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/layout/glass_activity_connect.xml:1]: nit: Missing header

File: WebRTCSample/app/res/drawable/ic_launcher_foreground.xml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (idzkowski-google) [inline WebRTCSample/app/res/drawable/ic_launcher_foreground.xml:1]: nit: Missing header

File: WebRTCSample/app/build.gradle [hunk 12]
```diff
@@ -0,0 +1,56 @@
+apply plugin: 'com.android.application'
```
Comment (idzkowski-google) [inline WebRTCSample/app/build.gradle:1]: nit: Missing header

File: WebRTCSample/app/AndroidManifest.xml [hunk 13]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (idzkowski-google) [inline WebRTCSample/app/AndroidManifest.xml:1]: nit: Missing header

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/FileDataSaver.java [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * Data saver that uses file to save data.
+     *
+     * @param fileCreator to create a file to save data into.
+     */
+    public FileDataSaver(FileCreator fileCreator) {
+        file = fileCreator.createFile();
+        separator = System.getProperty("line.separator");
+
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/FileDataSaver.java:29]: nit: Empty line

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/BaseCallActivity.java [hunk 15]
```diff
@@ -0,0 +1,868 @@
...
+import org.webrtc.RendererCommon.ScalingType;
+import org.webrtc.ScreenCapturerAndroid;
+import org.webrtc.SessionDescription;
+import org.webrtc.SurfaceViewRenderer;
+import org.webrtc.VideoCapturer;
+import org.webrtc.VideoFileRenderer;
+import org.webrtc.VideoFrame;
+import org.webrtc.VideoSink;
+
+public abstract class BaseCallActivity extends Activity implements AppRTCClient.SignalingEvents,
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/BaseCallActivity.java:52]: nit: Missing javadoc for public class

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/BaseConnectActivity.java [hunk 16]
```diff
@@ -0,0 +1,417 @@
...
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.util.Log;
+import android.webkit.URLUtil;
+import java.util.Random;
+
+
+public abstract class BaseConnectActivity extends Activity {
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/BaseConnectActivity.java:16]: nit: Missing javadoc for public class

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/FileSaverFactory.java [hunk 17]
```diff
@@ -0,0 +1,10 @@
+package org.appspot.apprtcstandalone;
+
+public class FileSaverFactory {
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/FileSaverFactory.java:3]: nit: Missing javadoc for public class

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+import android.app.FragmentTransaction;
+import android.os.Bundle;
+import android.util.DisplayMetrics;
+import android.view.MotionEvent;
+import com.example.glass.ui.GlassGestureDetector;
+import com.example.glass.ui.GlassGestureDetector.Gesture;
+import com.example.glass.ui.GlassGestureDetector.OnGestureListener;
+import org.webrtc.StatsReport;
+
+public class GlassCallActivity extends BaseCallActivity implements OnGestureListener {
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java:13]: nit: Missing javadoc for public class

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+import android.util.DisplayMetrics;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.SeekBar;
+import android.widget.TextView;
+import javax.annotation.Nullable;
+import org.webrtc.RendererCommon.ScalingType;
+
+public class GlassCallFragment extends Fragment {
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java:15]: nit: Missing javadoc for public class

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java [hunk 20]
```diff
@@ -0,0 +1,19 @@
+package org.appspot.apprtcstandalone;
+
+public class Resolution {
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java:3]: nit: Missing javadoc for public class

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java [hunk 21]
```diff
@@ -0,0 +1,22 @@
+package org.appspot.apprtcstandalone;
+
+public class SimpleCaptureOptions implements CaptureOptionsPicker.CaptureOptions {
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java:3]: nit: Missing javadoc for public class

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java [hunk 22]
```diff
@@ -0,0 +1,19 @@
+package org.appspot.apprtcstandalone.util;
+
+import java.util.Random;
+
+public class RandomRoomNameCreator implements RoomNameCreator {
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java:5]: nit: Missing javadoc for public class

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RoomNameCreator.java [hunk 23]
```diff
@@ -0,0 +1,5 @@
+package org.appspot.apprtcstandalone.util;
+
+public interface RoomNameCreator {
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RoomNameCreator.java:3]: nit: Missing javadoc for public interface

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+
+    private static final int NAME_LENGTH = 6;
+    private static final int VALUE_RANGE = 10;
+
+    @Override
+    public String getRoomName() {
+        Random random = new Random(System.currentTimeMillis());
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java:12]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java [hunk 25]
```diff
@@ -0,0 +1,19 @@
...
+        StringBuilder roomName = new StringBuilder();
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java:13]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+
+    private Resolution resolution;
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java:5]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java [hunk 27]
```diff
@@ -0,0 +1,22 @@
...
+    private int framerate;
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java:6]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java [hunk 28]
```diff
@@ -0,0 +1,79 @@
...
+    private final CharSequence fileName;
+
+    /**
+     * Default constructor
+     *
+     * @param directoryName directory name to be created in systemlogs directory.
+     * @param fileName      the name of the file to be appended with date of file creation.
+     */
+    public SdCardFileCreator(CharSequence directoryName, CharSequence fileName) {
+
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:30]: nit: Empty line

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java [hunk 29]
```diff
@@ -0,0 +1,79 @@
...
+        this.directoryName = directoryName;
+        this.fileName = fileName;
+    }
+
+    @Override
+    public File createFile() {
+        File directory =
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:37]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java [hunk 30]
```diff
@@ -0,0 +1,79 @@
...
+                new File(getFilePath(directoryName));
+        if (!directory.exists()) {
+            boolean mkdir = directory.mkdirs();
+            if (!mkdir) {
+                return null;
+            }
+        }
+
+        File file =
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:46]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java [hunk 31]
```diff
@@ -0,0 +1,79 @@
...
+            file.createNewFile();
+        } catch (IOException e) {
+            Log.e(TAG, "Cannot create log file.");
+            return null;
+        }
+        return file;
+    }
+
+    private String getCreationDate() {
+        StringBuilder outputDate = new StringBuilder();
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:63]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java [hunk 32]
```diff
@@ -0,0 +1,79 @@
...
+        Calendar cal = Calendar.getInstance();
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:64]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java [hunk 33]
```diff
@@ -0,0 +1,79 @@
...
+        outputDate.append(cal.get(Calendar.DAY_OF_MONTH)).append(SEPARATOR);
+        outputDate.append(cal.get(Calendar.HOUR_OF_DAY)).append(SEPARATOR);
+        outputDate.append(cal.get(Calendar.MINUTE)).append(SEPARATOR);
+        outputDate.append(cal.get(Calendar.SECOND));
+        return outputDate.toString();
+    }
+
+    private String getFilePath(CharSequence directory) {
+        return SD_CARD_DIR + SLASH + directory;
+
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:76]: nit: Empty line

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java [hunk 34]
```diff
@@ -0,0 +1,19 @@
...
+    private int height;
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java:4]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java [hunk 35]
```diff
@@ -0,0 +1,19 @@
...
+    private int width;
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java:5]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java [hunk 36]
```diff
@@ -0,0 +1,29 @@
...
+
+  public GlassHudFragment() {
+    super();
+  }
+
+  @Nullable
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View controlView = inflater.inflate(R.layout.glass_fragment_hud, container, false);
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java:24]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java [hunk 37]
```diff
@@ -0,0 +1,29 @@
...
+    TextView roomName = controlView.findViewById(R.id.room_name_textview);
+    roomName.setText(getArguments().getString(GlassCallActivity.EXTRA_ROOMID));
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java:26]: nit: Should be moved to the onViewCreated method

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java [hunk 38]
```diff
@@ -0,0 +1,29 @@
...
+  public GlassHudFragment() {
+    super();
+  }
+
+  @Nullable
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View controlView = inflater.inflate(R.layout.glass_fragment_hud, container, false);
+    TextView roomName = controlView.findViewById(R.id.room_name_textview);
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java:25]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java [hunk 39]
```diff
@@ -0,0 +1,140 @@
...
+      } else {
+        Log.e(TAG, "no permission to record audio/video");
+        finish();
+      }
+    }
+  }
+
+  private void resetViews() {
+    View mainView = getLayoutInflater().inflate(R.layout.glass_activity_connect, null, false);
+    setContentView(mainView);
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java:71]: nit: Should be moved to onCreate

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java [hunk 40]
```diff
@@ -0,0 +1,140 @@
...
+
+    gestureDetector = new GlassGestureDetector(this, this);
+    roomNameCreator = new RandomRoomNameCreator();
+    roomNameTextView = findViewById(R.id.room_name_textview);
+    roomNameTextView.setText(roomNameCreator.getRoomName());
+  }
+
+  private void refreshUi() {
+    isConnecting = true;
+    ViewGroup mainView = findViewById(android.R.id.content);
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java:81]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java [hunk 41]
```diff
@@ -0,0 +1,64 @@
...
+    View controlView = inflater.inflate(R.layout.glass_fragment_call, container, false);
+    captureFormatText = controlView.findViewById(R.id.capture_format_text_call);
+    captureFormatSlider = controlView.findViewById(R.id.capture_format_slider_call);
+    return controlView;
+  }
+
+  @Override
+  public void onStart() {
+    super.onStart();
+    DisplayMetrics displayMetrics = new DisplayMetrics();
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java:49]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java [hunk 42]
```diff
@@ -0,0 +1,64 @@
...
+    void onVideoScalingSwitch(ScalingType scalingType);
+    void onCaptureFormatChange(int width, int height, int framerate);
+    boolean onToggleMic();
+  }
+
+  @Nullable
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View controlView = inflater.inflate(R.layout.glass_fragment_call, container, false);
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java:40]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java [hunk 43]
```diff
@@ -0,0 +1,64 @@
...
+    captureFormatText = controlView.findViewById(R.id.capture_format_text_call);
+    captureFormatSlider = controlView.findViewById(R.id.capture_format_slider_call);
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java:42]: nit: Should be moved to the onViewCreated method

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java [hunk 44]
```diff
@@ -0,0 +1,62 @@
...
+
+  private GlassCallFragment callFragment;
+  private GlassHudFragment hudFragment;
+  private GlassGestureDetector gestureDetector;
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    DisplayMetrics displayMetrics = new DisplayMetrics();
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java:22]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java [hunk 45]
```diff
@@ -0,0 +1,62 @@
...
+
+    callFragment = new GlassCallFragment();
+    callFragment.setArguments(getIntent().getExtras());
+    hudFragment = new GlassHudFragment();
+    hudFragment.setArguments(getIntent().getExtras());
+
+    gestureDetector = new GlassGestureDetector(this, this);
+
+    callFragment.setArguments(getIntent().getExtras());
+    FragmentTransaction ft = getFragmentManager().beginTransaction();
```
Comment (idzkowski-google) [inline WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java:33]: nit: Can be final

File: WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java [hunk 46]
```diff
@@ -0,0 +1,140 @@
+package org.appspot.apprtcstandalone;
```
[the remark this answers is not in the document]
Comment (jbolinger) [reply WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java:16]: Added (and same for other `Glass*`) files. We shouldn't add headers to the existing code that was modified from the example.

**Comment by jbolinger** [comment]
Thanks @idzkowski-google - we don't want to mess with all of the code that was forked so I'll leave those unresolved. I fixed a few that were `Glass*` specific and added a copy of the original license to the readme so it's clearer.

**Pull request merged**